### PR TITLE
239 advanced search updates

### PR DESCRIPTION
**Pull request by cdmo**
cleared conflicts from #270

### CODE REVIEW

File: app/controllers/search_history_controller.rb [hunk 1]
```diff
@@ -3,5 +3,9 @@
 class SearchHistoryController < ApplicationController
   include Blacklight::SearchHistory
 
+  # TODO: Consider deleting the following three lines altogether. The following lines were generated during an install.
+  # Removing these lines has zero effect on the display of constraints.
```
Comment (banukutlu) [inline app/controllers/search_history_controller.rb]: thanks for noting this 👍

File: solr/conf/schema.xml [hunk 2]
```diff
@@ -270,8 +270,6 @@
   <copyField source="author_tsim" dest="author_spell"/>
   <copyField source="subject_tsim" dest="subject_spell"/>
   <copyField source="title_tsim" dest="title_spell"/>
-  <copyField source="pub_date_facet" dest="pub_date_ssm"/>
```
Comment (banukutlu) [inline solr/conf/schema.xml]: yay for removing unnecessary fields :)

**Comment by banukutlu** [comment]
I am getting this error on my local when I add a start or end date for publication year facet with branch 239 checked out.  try this [url](http://localhost:3000/?utf8=%E2%9C%93&range%5Bpub_date_sort_itsi%5D%5Bbegin%5D=&range%5Bpub_date_sort_itsi%5D%5Bend%5D=&search_field=all_fields&q=&range%5Bpub_date_sort_itsi%5D%5Bbegin%5D=1755&range%5Bpub_date_sort_itsi%5D%5Bend%5D=&commit=Limit). Maybe sth is up with my local, wondering if you are getting this as well.

Screen Shot 2019-04-19 at 10 22 39 PM

**Comment by cdmo** [comment]
That error happened because I made a mistake when manually correcting erb-lint violations.

Also, I think I prefer rebasing a feature being PR'd on top of `master` rather than merging `master` into a feature branch during a PR. That means that a PR has to be closed and reopened, but, I think it's cleaner. What do you think?

**Comment by banukutlu** [comment]
> That error happened because I made a mistake when manually correcting erb-lint violations.
> 
> Also, I think I prefer rebasing a feature being PR'd on top of `master` rather than merging `master` into a feature branch during a PR. That means that a PR has to be closed and reopened, but, I think it's cleaner. What do you think?

agreed. rebasing is cleaner 👍

**Pull request merged**